src/replication.c

```diff
@@ -1461,6 +1461,18 @@ void replconfCommand(client *c) {
                 return;
             }
             c->main_ch_client_id = (uint64_t)client_id;
+            /* Inherit the rdb-no-compress request from the main channel. */
+            if (main_ch && (main_ch->slave_req & SLAVE_REQ_RDB_NO_COMPRESS))
```

Comment:
```suggestion
            if (main_ch->slave_req & SLAVE_REQ_RDB_NO_COMPRESS)
```
it is up to you if you prefer being defensive but `main_ch` cannot be NULL at this point.